src/moonbeam/moonbase-local.js

```diff
@@ -283,7 +289,19 @@ const main = async () => {
         proxyAccountId,
     });
 
-    // TODO: need to add event verification of the ethereum execution
+    // Listen XCM events on Mangata side
+    const additionalWaitingTime = 5 * 60 * 1000;
+    const currentTimestamp = moment().valueOf();
+    const executionTime = (currentTimestamp - (currentTimestamp % millisecondsInHour)) + millisecondsInHour;
+    console.log(`\n5. Keep Listening XCM events on ${parachainName} until ${moment(executionTime).format('YYYY-MM-DD HH:mm:ss')}(${executionTime}) to verify that the task(taskId: ${taskId}, providerId: ${providedId}) will be successfully executed ...`);
+    const timeout = executionTime - currentTimestamp + additionalWaitingTime;
```

Comment:
It’s hard for a reader to understand the calculation here. moment library has an existing function to calculate the start of an hour, and looks more intuitive.

For example, next hour
`const timestampNextHour = (moment().add(1, 'hour').startOf('hour')).valueOf() / 1000;`

In two hours
`const timestampNextTwoHours = (moment().add(2, 'hour').startOf('hour')).valueOf() / 1000;`

Then, you can use the timestamp to construct the schedule parameter.
`const schedule = { Recurring: { frequency, nextExecutionTime: timestampNextHour } };`

Reply:
Fixed